formats/protobuf/src/test/java/io/cloudevents/protobuf/ProtobufFormatTest.java

```diff
@@ -131,6 +134,43 @@ public void roundTripTest(String filename) throws IOException {
 
     }
 
+    /**
+     * This test verifies the fix for Issue #523 that reported the Data
+     * portion was being corrupted during multi-step round trips if the
+     * data was a Protobuf Message.
+     */
+    @Test
+    public void verifyMultiRoundTrip() {
+
+        io.cloudevents.CloudEvent event = CloudEventBuilder.v1()
+            .withId(UUID.randomUUID().toString())
+            .withSource(SOURCE)
+            .withType(TYPE)
+            .withData(ProtoCloudEventData.wrap(io.cloudevents.test.v1.proto.Test.Quote.newBuilder()
+                .setHigh(io.cloudevents.test.v1.proto.Test.Decimal.newBuilder().setScale(2).setUnscaled(4200).build())
+                .setPrice(io.cloudevents.test.v1.proto.Test.Decimal.newBuilder().setScale(2).setUnscaled(1000).build())
+                .setSymbol("$")
+                .build()))
+            .build();
+
+
+
+        // 1st Round Trip
+        byte[] raw1 = format.serialize(event);
+        io.cloudevents.CloudEvent act1 = format.deserialize(raw1);
+        assertThat(event).withFailMessage("Mismatch on 1st round trip").isEqualTo(act1);
+
+        // 2nd Round Trip
+        byte[] raw2 = format.serialize(act1);
+        io.cloudevents.CloudEvent act2 = format.deserialize(raw2);
+        assertThat(event).withFailMessage("Mismatch on 2nd round trip").isEqualTo(act2);
+
+        // 3rd Time's a charm
+        byte[] raw3 = format.serialize(act2);
+        io.cloudevents.CloudEvent act3 = format.deserialize(raw3);
+        assertThat(event).withFailMessage("Mismatch on 3rd round trip").isEqualTo(act3);
```

Comment:
I would compare the actual bytes with each other in addition to the event

Reply:
This test is more around asserting that the CloudEvent Object is consistent across the various serialize/deserialize operations than the wire-format. 

If we wanted to add that level of check I'd suggest re-working the equals() implementation from your prior comment.

Reply:
Added new UnitTests to address byte[] equality.